renderdoc/driver/vulkan/vk_counters.cpp

```diff
@@ -85,6 +195,34 @@ CounterDescription VulkanReplay::DescribeCounter(GPUCounter counterID)
     }
   }
 
+  if(IsKHRCounter(counterID))
+  {
+    const VkPerformanceCounterKHR& khrCounter =
+      m_KHRCounters[FromKHRCounter(counterID)];
+    const VkPerformanceCounterDescriptionKHR& khrCounterDesc =
+      m_KHRCountersDescriptions[FromKHRCounter(counterID)];
+
+    CounterDescription rdcDesc;
+    rdcDesc.counter = counterID;
+    rdcDesc.name = khrCounterDesc.name;
+    rdcDesc.category = khrCounterDesc.category;
+    rdcDesc.description = khrCounterDesc.description;
+
+    auto unitIter = m_KHRCounterUnitsMap.find(khrCounter.unit);
+    if (unitIter != m_KHRCounterUnitsMap.end())
+    {
+        rdcDesc.unit = unitIter->second.unit;
+        rdcDesc.resultType = unitIter->second.type;
+        rdcDesc.resultByteWidth = unitIter->second.byteWidth;
+    }
+    else
+      RDCWARN("Unknown KHR counter unit %i, please fix Renderdoc.", khrCounter.unit);
+
+    // TODO: rdcDesc.uuid
```

Comment:
The UUID is important to support save/load of counters across sessions (since the counter ID may not remain constant if a driver adds a new counter). There is a `uuid` member in `VkPerformanceCounterKHR` which can be used directly I believe.

Reply:
How did you choose the uuid for the generic vulkan counters? :

```
desc.uuid.words[0] = 0x6839CB5B;                      
desc.uuid.words[1] = 0xFBD24550;                      
desc.uuid.words[2] = 0xB6068C65;                      
desc.uuid.words[3] = 0x157C684C ^ (uint32_t)counterID;
```

Reply:
It's a randomly generated UUID, but it only works because the counter IDs are fixed at compile time. As I say this field should be filled out with `VkPerformanceCounterKHR::uuid` and not with anything we generate ourselves. That one is a driver-provided UUID, so we can be sure that if we save a config referring to some counters it will work again later (on the same driver and similar enough hw to have that counter).

Reply:
Oops, forget that question... I didn't realize both uuid types matched in size.